xilem/src/lib.rs

```diff
@@ -180,25 +184,115 @@ pub struct Xilem<State, Logic> {
-    background_color: Color,
     // Font data to include in loading.
     fonts: Vec<Blob<u8>>,
 }
 
-impl<State, Logic, View> Xilem<State, Logic>
+/// State type used by [`Xilem::new_simple`].
+pub struct ExitOnClose<S> {
+    state: S,
+    running: bool,
+}
+
+impl<S> AppState for ExitOnClose<S> {
+    fn keep_running(&self) -> bool {
+        self.running
+    }
+}
+
+impl<State>
+    Xilem<
+        ExitOnClose<State>,
+        Box<
+            dyn FnMut(
+                &mut ExitOnClose<State>,
+            ) -> std::iter::Once<(
+                WindowId,
+                WindowOptions<ExitOnClose<State>>,
+                Box<AnyWidgetView<ExitOnClose<State>>>,
+            )>,
+        >,
+    >
+{
+    /// Create an app builder for a single window app with fixed window attributes
+    /// that exits once the window is closed.
+    ///
+    /// If you want to have multiple windows or change e.g. the window title depending
+    /// on the state you should instead use [`Xilem::new`] (which this function wraps).
+    pub fn new_simple<View>(
+        state: State,
+        mut logic: impl FnMut(&mut State) -> View + 'static,
+        window_options: WindowOptions<State>,
+    ) -> Self
+    where
+        View: WidgetView<State>,
+        State: 'static,
+    {
+        let window_id = WindowId::next();
+        let callbacks = Arc::new(window_options.callbacks);
+        Xilem::new_inner(
+            ExitOnClose {
+                state,
+                running: true,
+            },
+            Box::new(move |ExitOnClose { state, .. }| {
+                let callbacks = callbacks.clone();
+                let on_close = move |wrapper: &mut ExitOnClose<_>| {
+                    wrapper.running = false;
+                    if let Some(on_close) = &callbacks.on_close {
+                        on_close(&mut wrapper.state);
+                    }
+                };
+                std::iter::once((
+                    window_id,
+                    WindowOptions {
+                        reactive: window_options.reactive.clone(),
+                        initial: window_options.initial.clone(),
+                        callbacks: WindowCallbacks {
+                            on_close: Some(Box::new(on_close)),
+                        },
+                    },
+                    map_state(logic(state), |wrapper: &mut ExitOnClose<_>| {
+                        &mut wrapper.state
+                    })
+                    .boxed(),
+                ))
+            }),
+        )
+    }
+}
+
+/// The trait [`Xilem::new`] expects to be implemented for the state.
+///
+/// [`Xilem::new_simple`] does not use this trait implementation.
+pub trait AppState {
+    /// Returns whether the application should keep running or exit.
+    ///
+    /// Is currently only checked after a close request.
+    // TODO: check this after every state mutation
+    fn keep_running(&self) -> bool;
+}
```

Comment:
I suspect you could make your code a lot simpler by removing that trait and moving the method to `WindowView`.

Reply:
The intention behind this is that any callback (that can mutate the state) could quit the app so e.g. also the click callback of a button. I think deriving the "keep running" boolean from the app state is a nicer solution to this than passing some additional context to every single callback. With this in mind I don't think moving it into `WindowView` makes sense.